### DISCUSSION TITLE
Issues querying resources by resourceID

### DISCUSSION THREAD

**Discussion by mcdonamw** [Azure/CloudShell#280] [Q&A] [answered]
**UPDATE:**  I realized the help documentation I was looking at was for Az module version 9.3.0.  I'm using the Cloud Shell on the Azure portal which is apparently a lower version and that is the cause of the issue.  This works fine with Az 9.3.0 installed locally on Windows.

---------------------------------------------------------------------------

I'm trying to find Az resources using the Cloud Shell (Powershell) and it seems the command is not working as expected.

If I run Get-AzResource, and take the ResourceID from one of the results, and turn around retry that same query but specifying the -ResourceID property with said ResourceID, the command fails with an error I do not understand.

```

PS /home/me> Get-AzResourceGroup -Name 'rgTest' | select resourceID                                              

ResourceId
----------
/subscriptions/76ff5592-3577-488c-9d4c-455f06f7ab00/resourceGroups/rgTest

PS /home/me> get-azresource -resourceID '/subscriptions/76ff5592-3577-488c-9d4c-455f06f7ab00/resourceGroups/rgTest'                                                 
Get-AzResource: Invalid format of the resource identifier. (Parameter 'idFromServer')

PS /home/me> get-azresource -resourceID /subscriptions/76ff5592-3577-488c-9d4c-455f06f7ab00/resourceGroups/rgTest                                                 
Get-AzResource: Invalid format of the resource identifier. (Parameter 'idFromServer')
```

_**Note:**  Subscription GUID and resource have been modified for security._

From the documentation [See Here](https://learn.microsoft.com/en-us/powershell/module/az.resources/get-azresource?view=azps-9.3.0#example-7-get-a-resource-by-resource-id) my query is proper usage of this command so I do not understand what is going on.  Example provided:

```
Get-AzResource -ResourceId /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/testRG/providers/Microsoft.Compute/virtualMachines/testVM

Name              : testVM
ResourceGroupName : testRG
ResourceType      : Microsoft.Compute/virtualMachines
Location          : westus
ResourceId        : /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/testRG/providers/Microsoft.Compute/virtualMachines/testVM
Tags              :
                    Name    Value
                    ======  ========
                    Dept    IT
                    Year    2002
                    Status  Approved
```

Further issue:  As the example shows, one of the returned properties should be ResourceType, but I'm not showing this property being provided by the Get-AzResource command, currently.  This is the property I'm after.

Is this a bug?

**Comment by mcdonamw** [comment, marked as answer]
I realized the help documentation I was looking at was for Az module version 9.3.0. I'm using the Cloud Shell on the Azure portal which is apparently a lower version and that is the cause of the issue. This works fine with Az 9.3.0 installed locally on Windows.

**Comment by edyoung** [comment]
Cloud Shell is currently using 9.3.0. In general, release in Cloud Shell can't be at exactly the same instant as release to the module download location so Cloud Shell might be behind in some regions for a day or so post-release.
